### ISSUE TITLE
Automatically update previously-published Gists

### ISSUE THREAD

**Issue by devnoname120** [dnamsons/obsidian-publish-gist#13]
It would be super handy if gists could be updated automatically without requiring to manually run the command. You could perform the update when the editor is inactive for 30 seconds for example.

What do you think? ☺️

A ‶last updated″ frontmatter field and a ‶commit″ field would be helpful. The plugin could then display the sync status, just like Obsidian Sync does.

**Comment by dnamsons** [comment]
@devnoname120 I totally agree - this is one of the things that I'd like for this plugin to do.

However, I would like to approach this a bit differently - I think that it would be more valuable to create a some sort of "cache" that stores all of the notes that are currently uploaded to a gist and the last time each note was synced. 

This would allow for the plugin would always know of all the notes that should be kept in sync. This would also reduce the clutter in the frontmatter. I think that this approach could even remove the need for the `gist_id` property, as the id could be stored in the cache.

As for Obsidian Sync - I haven't used it and I do not know what the UI looks like. Do you have any screenshots that you could share so I could understand better?

**Comment by devnoname120** [comment]
> I think that this approach could even remove the need for the gist_id property, as the id could be stored in the cache.

I respectfully disagree about this one. It's super helpful to see in an eye blink where the file is getting synchronized to. Additionally, this avoids reconfiguring the plugin every time the vault is synced to another device (using Dropbox, Obsidian Sync, or something else.)

Tangentially, what would you think about switching to a `gist_url` property so that we could directly open the relevant gist by right-clicking on it? `obsidian-publish-gist` could then extract out the id from this url.

> As for Obsidian Sync - I haven't used it and I do not know what the UI looks like. Do you have any screenshots that you could share so I could understand better?

Sure! Here is a screenshot of the ‶sync status″ icon:

![image](https://user-images.githubusercontent.com/2824100/154753518-185f4d9c-c03f-48c8-8d2b-5a4025b6aaaf.png)

**Comment by dnamsons** [comment]
> Additionally, this avoids reconfiguring the plugin every time the vault is synced to another device 

The information on file syncronization statuses could be stored in a file within the `.obsidian` folder, therefore all the configuration would persist between different devices. 

I do, however agree about the usefulness of some kind of indication about where the note will be uploaded to.

As for the request for replacing the `gist_id` frontmatter property with `gist_url`, I created a seperate issue for that - #17

**Comment by devnoname120** [comment]
> The information on file syncronization statuses could be stored in a file within the .obsidian folder, therefore all the configuration would persist between different devices.

I'm not sure how well Obsidian Sync performs with this...

Alternatively we could add a `gist_commit_hash` and `gist_last_updated_at` to make synchronization stateless. It's hard as a user to figure out what's going on without having the information available right away.

Thanks to `gist_commit_hash`, you could warn the user if the Gist has been independently updated, and ask for confirmation to overwrite the changes or reconciliate them.

What do you think?

**Comment by devnoname120** [comment]
@dnamsons Friendly bump :)
